viewing_party/party.py

```diff
@@ -1,23 +1,156 @@
 # ------------- WAVE 1 --------------------
-
+import copy
+'''
+This function takes in 3 values: the title, genre, and rating of a movie. It uses these three parameters
+and creates a dictionary that includes the title, genre, and rating for the movie. Returns none if any of
+the parameters are "falsy".
+'''
 def create_movie(title, genre, rating):
-    pass
-
-# -----------------------------------------
-# ------------- WAVE 2 --------------------
-# -----------------------------------------
-
-
-# -----------------------------------------
-# ------------- WAVE 3 --------------------
-# -----------------------------------------
-
-        
-# -----------------------------------------
-# ------------- WAVE 4 --------------------
-# -----------------------------------------
-
-# -----------------------------------------
-# ------------- WAVE 5 --------------------
-# -----------------------------------------
-
+    if not title or not genre or not rating:
+        return None
+    movie = {
+        "title": title,
+        "genre": genre,
+        "rating": rating
+    }
+    return movie    
+'''
+This function takes in 2 values, a dictionary called user_data, and a dictionary called movie. It adds the movie
+as a value to the user_data under the key "watched", which is a list. 
+'''
+def add_to_watched(user_data, movie):
+    user_data_copy = copy.deepcopy(user_data)
+    user_data_copy["watched"].append(movie)
+    return user_data_copy
+'''
+This function takes in 2 values, a dictionary called user_data, and a dictionary called movie. It adds the movie
+as a value to the user_data under the key "watchlist", which is a list. 
+'''
+def add_to_watchlist(user_data, movie):
+    user_data_copy = copy.deepcopy(user_data)
+    user_data_copy["watchlist"].append(movie)
+    return user_data_copy
+'''
+This function takes in 2 values, a dictionary called user_data, and a string called title. It checks to make sure that the movie
+is in watchlist. If it is not in watchlist, we return user_data as is. If it is in watchlist, we will remove the movie from "watchlist" and 
+add it to "watched".
+'''
+def watch_movie(user_data, title):
+    user_data_copy = copy.deepcopy(user_data)
+    in_watchlist = False
+    movie_index = 0
+    for movie in user_data_copy["watchlist"]:
+        if movie["title"] == title:
+            in_watchlist = True

```

Comment:
You'll often see code avoid this additional boolean value by using a "sentinel" value for `movie_index` like `-1`. If you find it, the index is a valid index; if not, it's `-1`. You could use `None` as a sentinel, too.